### DISCUSSION TITLE
Mysterious rebasing ⁉

### DISCUSSION THREAD

**Discussion by antoniokunstek** [foivz/pop-app-backend#21] [Q&A]
Mate (@mislavmatijevic), I tried rebasing my feature branch with your feature branch and I have interesting problem. 
When I rebased half of your changes persist during rebasing half not.

For example: UserService does not have register user function. 
![image](https://github.com/foivz/pop-app-backend/assets/61520141/6c8dac81-603a-4b0a-a282-018d0040b8d9)

Also it appears I do not have your latest code changes. For example, SuccessResponse and ErrorResponse does not extend Response abstract class. 
![image](https://github.com/foivz/pop-app-backend/assets/61520141/4a4f2512-039f-4702-ab8c-dc8410435f49)

I mean I will delete this branch and start a new one from current develop branch, but I mean idk how this happened.
Maybe if you do, share your knowledge. I will really appreciate that. 🥺

**Comment by mislavmatijevic** [comment]
Oy, mate! Bugger, innit? You sure you pulled them branch before ya rebased, aye?

**Reply by antoniokunstek** [reply]
Yes, 100%. I literally created local branch so I can see code before accepting code review. So I tried rebasing locally. It have some merge conflicts but on UserController and UserControllerTest classes, so it should not have anything to do with service. 
But yeah, little confused.. got to be honest.  😖

**Reply by mislavmatijevic** [reply]
I really can't figure it out what happened, especially if you had the newest code. I'm not sure whether you had cherry-picked any of my commits last week, maybe that confused git?

Anyhow, the moral of the story is to always rebase atop of `develop` and to never use smaller branches unless we're working in parallel and there's an urgent need to share a functionality (and avoid a major merge conflict).

**Reply by antoniokunstek** [reply]
I will tell you what I did at the next coffee (☕ || 🍺 ), maybe I made some mistake (but, wright now, I can not figure out what went wrong here). Anyhow, many thanks for your help.

**Reply by mislavmatijevic** [reply]
Too bad, this remains a mistery for now.
